scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -361,7 +361,9 @@ private void checkNonIsolatedConstructsInTypeDefinition(TypeDefinitionNode typeD
                     List<Qualifier> typeDefQualifiers = typeDefinitionSymbol.qualifiers();
                     if (hasQualifier(typeDefQualifiers, PUBLIC_KEYWORD) &&
                             !hasQualifier(qualifiers, ISOLATED_KEYWORD)) {
-                        reportIssue(typeDefinitionNode, CoreRule.PUBLIC_NON_ISOLATED_OBJECT_CONSTRUCT);
+                        Optional<Token> visibilityQualifier = typeDefinitionNode.visibilityQualifier();
+                        assert visibilityQualifier.isPresent();
```

Comment:
We can use this `visibilityQualifier` instead of 361.
any way im +1 for this as well